test/services/step_manager_service_test.rb

```diff
@@ -4,21 +4,37 @@
 
 class StepManagerServiceTest < ActiveSupport::TestCase
```

Comment:
@mark-cooper I added some tests here for public methods that I had added (and some behavior I changed). But a fair amount of the mess I added in here manipulates CSVs behind the scenes to generate the final report for each step without storing individual CSV row data in the database or caching it. Testing this stuff requires actual data in an input CSV associated with a batch, and the ability to verify the resulting CSV(s) are as expected in the tests. The fixtures as currently set up don't seem to support testing this and I was unable to figure out how to fix that today.